new/missing-method-missing.md

```diff
@@ -0,0 +1,156 @@
+`Chef::Resource::MyResource`.
+
+This means that if you have seen examples of classes in Chef::Resource, and want
+to do the obvious thing and just move to your own namespace, it doesn't
+immediately work and it is far from apparent why. The result? A massive traffic
+jam of non-core resources in the `Chef::Resource` namespace and puzzling error
+messages for cookbook conflicts such as "superclass mismatch for class MyResource."
+
+Chef will now automatically call `provides` on all new, named resource classes,
+using `ResourceClass.dsl_name` (a transform of the class name, i.e.
+MyPackage::MyResource -> my_resource) as the provided class.
+
+### Move LWRPs out of the Chef::Resource namespace
+
+With Chef::Resource no longer a special place for class lookup, LWRPs no longer
+need to be given a namespace.  We will now make resources/x.rb and providers/x.rb
+an anonymous class, with string methods to make it easy to see where they come
+from.  This avoids a number of potential errors and warnings that can happen in
+the case of conflicts, and gives us control over how we handle conflicts.
+
+Users who wish to set the class of the resource can assign it directly in the
+LWRP with `Namespace::MyResource = self`.
+
+## Deprecations and Errors
+
+Removing `method_missing` means several methods of creating resource names cannot
+be caught.  These are the cases.  We will issue deprecation warnings for each one
+and remove them (and method_missing) come Chef 13.
+
+### Chef::Resource::MyResource = <resource>
+
+When you assign an anonymous class or a resource from another namespace into
+`Chef::Resource`, your new resource will currently be accessible via the DSL.
+Since there is no Ruby method that lets us catch the assignment of a constant,
+there is no way for us to assign an explicit DSL method to handle the resource.
+
+```ruby
+# Deprecated
+Chef::Resource::Blah = Class.new(Chef::Resource)
+```
+
+We will catch this in method_missing, issue a warning letting the user know they
+should use the `provides` directive on their class to get it into the DSL, and remove it come Chef 13.
+
+```ruby
+# Not deprecated (though the assignment becomes unnecessary)
+Chef::Resource::Blah = Class.new(Chef::Resource) { provides :blah }
+```
+
+### Calling `method_missing` directly
+
+If people call `method_missing` directly to invoke DSL methods, and we don't
+define it anymore, it obviously won't actually invoke the methods.  We will
+detect this in method_missing, issue a warning telling people to call the
+actual method instead or use public_send(), and remove it come Chef 13.
+
+### Custom `provides?` without corresponding `provides :name`
+
+If someone creates a resource class with a custom `provides?` method, they can
+currently
+
+```ruby
+class MyResource < Chef::Resource
+  def provides?(node, name)
+    name == :blah
+  end
+end
+```
+
+The implications of this are that we have to scan every single resource when
```

Comment:
Oh, you mean that I should point out that each of the things I just called deprecations are compat breaks?  Isn't that sort of implied?  I can make the change, but just in case this comment was due to me not making myself clear enough, you could read the current verbage like this:

> In Chef 12 this will still work and cause a deprecation warning, and in Chef 13 the behavior will simply fail"?

Again, happy to move deprecation to 13 and break to 14, just wanted to make sure we're on the same page here regarding the nature of this RFC.  The intent is to break nothing at all in version N, but issue deprecation warnings for every behavior that will break in version N+1.